### ISSUE TITLE
Issue with TypeScript and custom props

### ISSUE THREAD

**Issue by daviddelusenet** [morajabi/styled-media-query#22]
Been using this package for multiple projects and really like it. However, in my latest project I'm using TypeScript and because of this I found a bug in `styled-media-query`.

I got a styled components which expects a `isVisible` boolean prop:

```ts
interface OverlayProps {
    isVisible: boolean;
}

export const Overlay = styled.div<OverlayProps>`
    color: ${({ isVisible }): string => (isVisible ? 'red' : 'green')};
`;
```

This all works fine and TypeScript doesn't throw any errors. However, if I try to use the `isVisible` prop inside `media` I'm getting errors:

```ts
interface OverlayProps {
    isVisible: boolean;
}

export const Overlay = styled.div<OverlayProps>`
    opacity: ${({ isVisible }): string => (isVisible ? '0.3' : '0')};

    ${media.lessThan('large')`
        background-color: ${({ theme }): string => theme.shades.one};
        position: fixed;
        top: 52px;
        left: ${({ isVisible }): string => (isVisible ? '260px' : '0')};
        width: 100%;
        height: 100%;
        z-index: 2;
        opacity: ${({ isVisible }): string => (isVisible ? '0.3' : '0')};
        cursor: pointer;
        transition: all 300ms ease;
    `}
`;
```

The `left` and `opacity` declarations are throwing the following error: `Property 'isVisible' does not exist on type 'ThemeProps<any>'`.

Screenshot: 

<img width="844" alt="Screenshot 2019-12-17 at 09 46 31" src="https://user-images.githubusercontent.com/1398418/70979553-1f52a980-20b2-11ea-8a5a-d7f9aa4fde22.png">

Please let me know if you need any more information. Any help would be greatly appreciated!

**Comment by daviddelusenet** [comment]
Any idea how to fix this? This is really painful for my current project so I would like to fix it asap!

If you need any more context or information please let me know.

**Comment by daviddelusenet** [comment]
I'm still encountering this 'bug' daily so I would really like to solve it..

**Comment by ysKuga** [comment]
Setting generics parameter to `media.lessThan` worked.
media.lessThan`<OverlayProps>`('large')

```tsx

interface OverlayProps {
  isVisible: boolean;
}

export const Overlay = styled.div<OverlayProps>`
  opacity: ${({ isVisible }): string => (isVisible ? '0.3' : '0')};

  ${media.lessThan<OverlayProps>('large')`
      background-color: ${({ theme }): string => theme.shades.one};
      position: fixed;
      top: 52px;
      left: ${({ isVisible }): string => (isVisible ? '260px' : '0')};
      width: 100%;
      height: 100%;
      z-index: 2;
      opacity: ${({ isVisible }): string => (isVisible ? '0.3' : '0')};
      cursor: pointer;
      transition: all 300ms ease;
  `}
`
```

![test](https://user-images.githubusercontent.com/17670283/72473266-a61db200-3829-11ea-8fc7-f656224ccf99.png)
